plugins/README.md

```diff
@@ -83,6 +83,10 @@ geometry_prompt_pretty_git_render() {
 }
 ```
 
+As a convention plugins should only be rendered when sitting on a valid context,
+for example the `node` built-in plugin will only be displayed when sitting on a
+npm/yarn-based project. This is done in order to have a uncluttered prompt.
```

Comment:
I think we could be a bit more clear on what we want users to follow. Something like "the `node` built-in plugin will check if a `package.json` or `yarn.lock` file is present before rendering. This means it will only be displayed when sitting (...). When developing a custom plugin, we encourage you to have this in mind."

Or equivalent.